### DISCUSSION TITLE
How to PROPERLY tween camera to player's pos

### DISCUSSION THREAD

**Discussion by NikRetaNCAM** [HaxeFlixel/flixel#2522] [Q&A]
Dumb question, again, sorry 😅 
When player finishes a level, i want camera to zoom to him and fade before transitioning to next level
![image](https://user-images.githubusercontent.com/60806448/159274250-2b65f71d-34fb-4784-a6aa-f888e840ef81.png)
This code just fades the camera and zooms it when player gets to finish point but now on the player's position
IF i do .tween(FlxG.camera, {x: player.x, y: player.y}) it will fly to left-up corner

Oh and about finish point, when i touch it
![image](https://user-images.githubusercontent.com/60806448/159274827-526ec7ff-95a7-4050-967a-b8942ef76f4c.png)
(i made it a bit upper on purpose)
It will fly to direction i touched it (if you go from the left, it will fly to the right, etc)
![image](https://user-images.githubusercontent.com/60806448/159275241-4bef410e-55ff-4a99-9537-f00e4255a85f.png)
Any ideas why that happens?

Thanks

**Comment by Geokureli** [comment]
If I understand the problem correctly, the goal is moving right when you enter from the left? does this happen when you comment out the tween code? I assume the issue has nothing to do with the tween, but that you're checking overlap between the player and end goal using `FlxG.collide` rather than `FlxG.overlap`. collide will check overlap and then attempt to separate any overlapping objects. use `FlxG.overlap

**Reply by NikRetaNCAM** [reply]
the goal is to prevent the finish point from moving. it shouldn't be moving, like, it's sort of altar or rune. And another goal is to get camera zooming to player's position when player touches finish point

These are kind of 2 separate tasks, ig

**Reply by Geokureli** [reply]
gotcha, first things first, did my advice to use FlxG.overlap work to stop it from moving?

**Reply by NikRetaNCAM** [reply]
> gotcha, first things first, did my advice to use FlxG.overlap work to stop it from moving?

yes, thanks for advice

**Reply by Geokureli** [reply]
cool! now, with zoom tweens they work for me because I tend to use `camera.follow(player); which keeps the camera pointing at my  otherwise you may need to tween camera.scroll.x and y to adjust

**Comment by Wither362** [comment]
`tween(FlxG.camera, {x: player.x + (player.width / 2), y: player.y + (player.height / 2});`
